modules/akka/src/main/scala/ch/epfl/bluebrain/nexus/sourcing/akka/SourcingConfig.scala

```diff
@@ -0,0 +1,123 @@
+package ch.epfl.bluebrain.nexus.sourcing.akka
+
+import akka.actor.ActorSystem
+import akka.util.Timeout
+import cats.ApplicativeError
+import cats.effect.Timer
+import ch.epfl.bluebrain.nexus.sourcing.akka.SourcingConfig._
+
+import scala.concurrent.ExecutionContext
+import scala.concurrent.duration.FiniteDuration
+
+/**
+  * Sourcing configuration.
+  *
+  * @param askTimeout                        timeout for the message exchange with the aggregate actor
+  * @param queryJournalPlugin                the query (read) plugin journal id
+  * @param commandEvaluationTimeout          timeout for evaluating commands
+  * @param commandEvaluationExecutionContext the execution context where commands are to be evaluated
+  * @param shards                            the number of shards for the aggregate
+  * @param passivation                       the passivation strategy configuration
+  * @param retry                             the retry strategy configuration
+  */
+final case class SourcingConfig(
+    askTimeout: FiniteDuration,
+    queryJournalPlugin: String,
+    commandEvaluationTimeout: FiniteDuration,
+    commandEvaluationExecutionContext: String,
+    shards: Int,
+    passivation: PassivationStrategyConfig,
+    retry: RetryStrategyConfig,
+) {
+
+  /**
+    * Computes an [[AkkaSourcingConfig]] using an implicitly available actor system.
+    *
+    * @param as the underlying actor system
+    */
+  def akkaSourcingConfig(implicit as: ActorSystem): AkkaSourcingConfig =
+    AkkaSourcingConfig(
+      askTimeout = Timeout(askTimeout),
+      readJournalPluginId = queryJournalPlugin,
+      commandEvaluationMaxDuration = commandEvaluationTimeout,
+      commandEvaluationExecutionContext =
+        if (commandEvaluationExecutionContext == "akka") as.dispatcher
+        else ExecutionContext.global
+    )
+
+  /**
+    * Computes a passivation strategy from the provided configuration and the passivation evaluation function.
+    *
+    * @param shouldPassivate whether aggregate should passivate after a message exchange
+    * @tparam State   the type of the aggregate state
+    * @tparam Command the type of the aggregate command
+    */
+  def passivationStrategy[State, Command](
+      shouldPassivate: (String, String, State, Option[Command]) => Boolean =
+        (_: String, _: String, _: State, _: Option[Command]) => false
+  ): PassivationStrategy[State, Command] =
+    PassivationStrategy(
+      passivation.lapsedSinceLastInteraction,
+      passivation.lapsedSinceRecoveryCompleted,
+      shouldPassivate
+    )
+}
+
+object SourcingConfig {
```

Comment:
Since we're always constructing these configs using `pureconfig`, I don't really see any use of an apply method other than the default one provided by the case class.

Reply:
I'm happy with the test dependency on pureconfig with a proper example.